python/cuda_cooperative/tests/test_block_reduce.py

```diff
@@ -214,6 +214,139 @@ def kernel(input, output):
     assert "STL" not in sass
 
 
+@pytest.mark.parametrize("T", [types.uint32, types.uint64])
+@pytest.mark.parametrize("threads_in_block", [32, 64, 128, 256, 512, 1024])
+def test_block_reduction_valid(T, threads_in_block):
+    def op(a, b):
+        return a if a < b else b
+
+    block_reduce = cudax.block.reduce(
+        dtype=T, binary_op=op, threads_in_block=threads_in_block
+    )
+    temp_storage_bytes = block_reduce.temp_storage_bytes
+
+    @cuda.jit(link=block_reduce.files)
+    def kernel(input, output):
+        temp_storage = cuda.shared.array(shape=temp_storage_bytes, dtype="uint8")
+        block_output = block_reduce(
+            temp_storage, input[cuda.threadIdx.x], threads_in_block / 2
+        )
+
+        if cuda.threadIdx.x == 0:
+            output[0] = block_output
+
+    dtype = NUMBA_TYPES_TO_NP[T]
+    h_input = random_int(threads_in_block, dtype)
```

Comment:
Does it make sense to have a zero at the end of  `h_input` to make sure we're not reducing across the whole block?

Reply:
Done.